### PR TITLE
default location barcode

### PR DESCRIPTION
**Pull request by Whenrow**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mrp/models/stock_warehouse.py [hunk 1]
```diff
@@ -170,9 +170,21 @@ def _get_locations_values(self, vals):
...
+        code = vals.get('code') or self.code
+        code = code.replace(' ', '').upper()
         values.update({
-            'pbm_loc_id': {'name': _('Pre-Production'), 'active': manufacture_steps in ('pbm', 'pbm_sam'), 'usage': 'internal'},
-            'sam_loc_id': {'name': _('Post-Production'), 'active': manufacture_steps == 'pbm_sam', 'usage': 'internal'},
+            'pbm_loc_id': {
+                'name': _('Pre-Production'),
+                'active': manufacture_steps in ('pbm', 'pbm_sam'),
+                'usage': 'internal',
+                'barcode': code + '-PREPRODUCTION'
```
Comment (Yenthe666) [inline addons/mrp/models/stock_warehouse.py]: @Whenrow perhaps nice that "PREPRODUCTION" is also translatable?

File: addons/mrp/models/stock_warehouse.py [hunk 2]
```diff
@@ -170,9 +170,21 @@ def _get_locations_values(self, vals):
...
+            },
+            'sam_loc_id': {
+                'name': _('Post-Production'),
+                'active': manufacture_steps == 'pbm_sam',
+                'usage': 'internal',
+                'barcode': code + '-POSTPRODUCTION'
```
Comment (Yenthe666) [inline addons/mrp/models/stock_warehouse.py]: @Whenrow perhaps nice that "PREPRODUCTION" is also translatable?

File: addons/stock/models/stock_warehouse.py [hunk 3]
```diff
@@ -439,12 +439,39 @@ def _get_locations_values(self, vals):
...
-            'lot_stock_id': {'name': _('Stock'), 'active': True, 'usage': 'internal'},
-            'wh_input_stock_loc_id': {'name': _('Input'), 'active': reception_steps != 'one_step', 'usage': 'internal'},
-            'wh_qc_stock_loc_id': {'name': _('Quality Control'), 'active': reception_steps == 'three_steps', 'usage': 'internal'},
-            'wh_output_stock_loc_id': {'name': _('Output'), 'active': delivery_steps != 'ship_only', 'usage': 'internal'},
-            'wh_pack_stock_loc_id': {'name': _('Packing Zone'), 'active': delivery_steps == 'pick_pack_ship', 'usage': 'internal'},
+            'lot_stock_id': {
+                'name': _('Stock'),
+                'active': True,
+                'usage': 'internal',
+                'barcode': code + '-STOCK'
```
Comment (Yenthe666) [inline addons/stock/models/stock_warehouse.py]: @Whenrow perhaps nice that "-STOCK" is also translatable?

Comment (Yenthe666) [reply addons/stock/models/stock_warehouse.py]: & same for those others lower in the file.

File: addons/mrp/models/stock_warehouse.py (hunk 2 above)
Comment (mreficent) [reply addons/mrp/models/stock_warehouse.py]: post

**Comment by sle-odoo** [comment]
robodoo r+

**Pull request closed without merging**